discovery/marathon/marathon.go

```diff
@@ -458,7 +459,7 @@ func targetsForApp(app *App) []model.LabelSet {
 
 			// A zero port can appear in a portMapping, which means it is auto-generated.
 			// The allocated port appears at the corresponding index in the task's 'ports' array.
-			if port == 0 && len(t.Ports) == len(ports) {
+			if (!app.RequirePorts || port == 0) && len(t.Ports) == len(ports) {
```

Comment:
Undo this change, an upstream zero port will cause this code to do The Right Thing. (tm)

Reply:
Awthanks!